src/NJsonSchema/Generation/JsonSchemaGenerator.cs

```diff
@@ -988,6 +988,26 @@ private void GenerateKnownTypes(Type type, JsonSchemaResolver schemaResolver)
 
             if (Settings.GenerateKnownTypes)
             {
+
+#if NET7_0_OR_GREATER
```

Comment:
should we also support NET 6? NET 6 with STJ 7.0 installed is also a valid combination, and I verified in my projects that `JsonPolymorphic` also works for STJ 7 under NET 6.

Reply:
Or maybe we could define a new constant `STJ7_0_OR_GREATER` use that as the condition

Reply:
> should we also support NET 6? NET 6 with STJ 7.0 installed is also a valid combination, and I verified in my projects that JsonPolymorphic also works for STJ 7 under NET 6.

Ah, you're right, the STJ 7.0.2 NuGet package supports `net6.0`. [It seems the docs aren't 100% accurate](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/polymorphism?pivots=dotnet-6-0): _In versions prior to .NET 7, System.Text.Json doesn't support the serialization of polymorphic type hierarchies._

> Or maybe we could define a new constant `STJ7_0_OR_GREATER` and use that as the condition

How would that work? Wouldn't the value of the directive depend on the .NET runtime version and also possibly an STJ NuGet dependency? I have no clue how to do that...

My idea was to focus on backwards compatibility. I'm not sure how common the usecase of .NET 6 + STJ 7 is. I can't say if it's worth the effort to support that scenario. If we want to support that, it's probably easier to just include the STJ 7 NuGet package for `net6.0`, `net462`, and `netstandard2.0`.

Reply:
>If we want to support that, it's probably easier to just include the STJ 7 NuGet package for net6.0, net462, and netstandard2.0.

hmmm I’m wondering… in this case, is this possible just to get rid of the NET7_0_OR_GREATER macro so that older .NET versions could leverage your work as well?

Reply:
I may need to take a closer look at the changes again, if getting rid of that macro is not possible/preferred. In general, I’m just trying to figure out a way to make your change available to net6 version or whichever version that STJ 7 supports.

Reply:
>Ah, you're right, the STJ 7.0.2 NuGet package supports net6.0. [It seems the docs aren't 100% accurate](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/polymorphism?pivots=dotnet-6-0): In versions prior to .NET 7, System.Text.Json doesn't support the serialization of polymorphic type hierarchies.

Yeah I agree… I’m thinking maybe what that doc wants to tell us is that for NET7 and above, STJ 7 is NATIVELY supported, which doesn’t necessarily mean that STJ 7 cannot be used in older NET versions.